IdentityCore/src/cache/key/MSIDMacKeychainAccountCacheKey.m

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "MSIDMacKeychainAccountCacheKey.h"
+
+static NSString *keyDelimiter = @"-";
+static NSInteger kAccountTypePrefix = 1000;
+
+@implementation MSIDMacKeychainAccountCacheKey
+
+- (instancetype)initWithHomeAccountId:(NSString *)homeAccountId
+                          environment:(NSString *)environment
+                                realm:(NSString *)realm
+                                 type:(nullable NSNumber *)type
+{
+    self = [super init];
+    
+    if (self)
+    {
+        _homeAccountId = homeAccountId;
+        _environment = environment;
+        _realm = realm;
+        _accountType = type;
+    }
+    
+    return self;
+}
+
+- (NSData *)generic
+{
+    return [self.username.msidTrimmedString.lowercaseString dataUsingEncoding:NSUTF8StringEncoding];
+}
+
+- (NSNumber *)type
+{
+    if (_accountType)
+    {
+        return @([_accountType integerValue] + kAccountTypePrefix);
+    }
+    else
+    {
+        return nil;
+    }
+}
+
+- (NSString *)account
+{
+    NSString *uniqueId = self.homeAccountId.msidTrimmedString.lowercaseString;
+    
+    return [NSString stringWithFormat:@"%@%@%@",
+            uniqueId, keyDelimiter, self.environment.msidTrimmedString.lowercaseString];
+}
+
+- (NSString *)service
+{
+    return self.realm.msidTrimmedString.lowercaseString;
+}
+
```

Comment:
```
/* Application key for keychain isolation */
@property (nullable, readwrite) NSString *appKey;
```

Why didn't you overload this one?

Reply:
It's in the MSIDCacheKey base class, and doesn't seem to be needed for accounts.